google-cloud-logging/src/test/java/com/google/cloud/logging/BaseSystemTest.java

```diff
@@ -273,17 +284,29 @@ public void testWriteAndListLogEntries() throws InterruptedException {
     while (iterator.hasNext()) {
       assertTrue(iterator.next().getTimestamp() <= lastTimestamp);
     }
-    assertTrue(logging().deleteLog(logId));
+    int deleteAttempts = 0;
+    int allowedDeleteAttempts = 5;
+    boolean deleted = logging().deleteLog(logId);
+    while (!deleted && deleteAttempts < allowedDeleteAttempts) {
+      deleteAttempts++;
+      System.out.println("Not deleted - " + deleteAttempts + " of "
+          + allowedDeleteAttempts + " attempted so far.");
+      Thread.sleep(1000);
+      deleted = logging().deleteLog(logId);
+    }
+    assertTrue(deleted);
```

Comment:
I think we should use `rateLimiter.acquire()` instead of `sleep`.
I also don't think it's necessary to print, but if you think that's worthwhile, then I think `System.err` is a better choice.

This might just be me; I find it easier to do
```java
boolean deleted = false;
while (...) {
  rateLimiter.acquire();
  deleted = logging().deleteLog(logId);
}
```
so we only need to write the delete once. Up to you though.